docs/ZMI.rst

```diff
@@ -98,3 +98,157 @@ The properties can have one of the following types:
 The value of the property has to be one or more paths/URLs to CSS resp.
 JavaScript which will be included in the HTML of the ZMI. (Paths have to be
 resolvable by the browser aka not simple file system paths.)
+
+
+Update your Zope2 ZMI template code
++++++++++++++++++++++++++++++++++++
+
+The Zope4 ZMI uses some basic CSS class names of the bootstrap cCSS framework 
+and structuring concepts for page layout and forms. A ZMI page usually sequences 
+following templates nesting the page core:
+
+1. manage_page_header()
+2. manage_tabs()
+3. page core
+4. manage_page_footer()
+
+The page core of any form or object listing ZMI template is starting by the 
+html element ``<main class="container-fluid">``.
+Ususally <main> is nesting a p-element for a help-text and the actual form. 
+To make specific form styling possible the form-element has following CSS names:
+
+1. zmi-$classname
+2. zmi-edit|-add
+
+
+In general specific functional ZMI elements are declared by a CSS class with a
+prefixed ``zmi-`` whereas the basic layout is done by usual bootstrap classes 
+fowllowing the typical element nesting:
+any form element has a bootstrap-like form-group structure containing a label 
+and an input field. Important: the width of the input field is defined by the 
+nesting div-container using the responsive grid classes ``col-sm-9 col-and md-10``. 
+The input field width is completed to 12 (=100%) by the preceeding label classes 
+``col-sm-3`` and ``col-md-2``
+
+.. raw:: html
+	<div class="form-group row">
+		<label for="title" class="form-label col-sm-3 col-md-2">Title</label>
+		<div class="col-sm-9 col-md-10">
+			<input id="title" class="form-control" type="text" name="title" 
+				value="<dtml-if title>&dtml-title;</dtml-if>" />
+		</div>
+	</div>
+
+The following buttons are constructed as div element with the classname 
+``zmi-controls``; the bbuttons use systematically the  bootstrap class pair 
+``btn btn-primary``.
+
+.. raw:: html
+	<div class="zmi-controls">
+		<input class="btn btn-primary" type="submit" name="submit" value="Save" />
+	</div
+
+The following example code shows a whole restructed DTML template rendering the 
+Zope4 ZMI
+
+**Example: updated DTML template** 
+(from: ``../Zope/src/OFS/dtml/documentEdit.dtml``)
+
+.. raw:: html
+	<dtml-var manage_page_header>
```

Comment:
Inlining the file (and thus keeping it simple, even if the actual template becomes more complex) can be valuable - but always showing the actual used code might also be.

If wanted, can Sphinx accept

```
.. raw:: html
   :file: filename.html
```

To include the actual template used?

Not sure this is a good idea.